python/ray/serve/controller.py

```diff
@@ -243,3 +243,18 @@ async def shutdown(self) -> None:
                 for replica in replica_dict.values():
                     ray.kill(replica, no_restart=True)
             self.kv_store.delete(CHECKPOINT_KEY)
+
+    async def deploy(self, name: str, backend_config: BackendConfig,
+                     replica_config: ReplicaConfig,
+                     version: Optional[str]) -> Optional[GoalId]:
+        """TODO."""
+        async with self.write_lock:
+            goal_id = self.backend_state.create_backend(
+                name, backend_config, replica_config, version)
+
+            self.endpoint_state.create_endpoint(name, f"/{name}",
+                                                ["GET", "POST"],
```

Comment:
Is it worth adding "POST" to the default methods in `serve.create_endpoint` as well for consistency? Right now it's just "GET", but adding "POST" might make it more user-friendly (https://discuss.ray.io/t/post-method-using-http-request-with-ray-serve/1250).

Reply:
I think with the new fastapi-native ingress support that @simon-mo is adding we can just let the application decide which methods it handles and we transparently forward them all through